parameterize simplification of line features in read_shp

chrisnatali opened the pull request:
I needed the segments of multi-segment line features to be added as edges, so I added a simplify parameter to read_shp (defaults to True so it's back compatible).  A caveat is that attributes associated with the feature are repeated for each segment.  Thought someone else might find this useful.

chebee7i commented:
Seems helpful. Could you add a unit test that covers this case? You should be able to modify one of the tests that appears in `networkx/readwrite/tests/test_shp.py`.

chebee7i commented on networkx/readwrite/nx_shp.py:
```diff
@@ -38,6 +38,11 @@ def read_shp(path):
+        If False, and line feature geometry has multiple segments, the 
+        attributes for that feature will be repeated for each edge comprising
+        that feature
+
```
It should be:
```
`simplify : bool
    If ``True``, simplify line geometries to the start and end 
    coordinates. If ``False``... comprising that feature.
```

chebee7i commented on networkx/readwrite/nx_shp.py:
```diff
@@ -73,8 +78,12 @@ def read_shp(path):
+                    last = g.GetPointCount() - 1
+                    net.add_edge(g.GetPoint_2D(0), g.GetPoint_2D(last), attributes)
+                else:
+                    for i in range(0, g.GetPointCount() - 1):
```
Might as well change this at the same time: The start integer 0 is typically not included in `range`. So just `range(g.GetPointCount() - 1)` should do.

SanketDG replied:
The `last` variable should be declared after line 80 where it previously was. Then line 85 should be changed to 
```python
for i in range(last):

Don't understand why variable `last` is inside the `if` block.

chrisnatali commented:
Thanks for the feedback.  Addressed comments.  Also for the ``simplify=False`` case it seemed consistent to "segmentize" the "Wkb", "Wkt" and "Json" attributes.
